### PR TITLE
Added Controllers and CRDs based on DBaaS operator design

### PR DESCRIPTION
**Pull request by redhatHameed**
What : 
This PR opened against issue #7 created CRDs and controller based on a design from DBaaS as shown below 
Screenshot from 2021-06-01 17-20-09

How :

1. Regenerated all controller/boilerplate code using the below commands 
```
operator-sdk init --domain crunchydata.com --repo github.com/CrunchyData/crunchy-bridge-operator
operator-sdk edit --multigroup=true
operator-sdk create api --group=crunchybridge --version=v1  --kind=BridgeCluster  --resource  --controller
operator-sdk create api --group dbaas.redhat.com  --version v1alpha1 --kind CrunchyBridgeInventory --resource --controller
operator-sdk create api --group dbaas.redhat.com  --version v1alpha1 --kind CrunchyBridgeConnection --resource --controller
```
2. updated types and CRDs for CrunchyBridgeInventory and CrunchyBridgeConnection

3. Regenerated bundles using `make bundle`  

Verify The operator using: `make run`
```
I0603 21:13:36.091228       1 request.go:655] Throttling request took 1.001202601s, request: GET:https://172.30.0.1:443/apis/crunchybridge.com/v1?timeout=32s
2021-06-03T21:13:41.888Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": "127.0.0.1:8080"}
2021-06-03T21:13:41.889Z	INFO	setup	starting manager
2021-06-03T21:13:41.889Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
I0603 21:13:41.889892       1 leaderelection.go:243] attempting to acquire leader lease crunchy-bridge-operator-system/0b67260c.crunchydata.com...
I0603 21:13:41.917859       1 leaderelection.go:253] successfully acquired lease crunchy-bridge-operator-system/0b67260c.crunchydata.com
2021-06-03T21:13:41.917Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"ConfigMap","namespace":"crunchy-bridge-operator-system","name":"0b67260c.crunchydata.com","uid":"f796f987-c63d-4c06-8ace-dede3ea51490","apiVersion":"v1","resourceVersion":"555417668"}, "reason": "LeaderElection", "message": "crunchy-bridge-operator-controller-manager-66ffc8c57d-m82g8_f724305f-754b-47fd-a7a7-01fe80205b4a became leader"}
2021-06-03T21:13:41.918Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"Lease","namespace":"crunchy-bridge-operator-system","name":"0b67260c.crunchydata.com","uid":"a9dfd89e-a9ce-40d2-805d-e76b55a19798","apiVersion":"coordination.k8s.io/v1","resourceVersion":"555417669"}, "reason": "LeaderElection", "message": "crunchy-bridge-operator-controller-manager-66ffc8c57d-m82g8_f724305f-754b-47fd-a7a7-01fe80205b4a became leader"}
2021-06-03T21:13:41.918Z	INFO	controller-runtime.manager.controller.bridgecluster	Starting EventSource	{"reconciler group": "crunchybridge.crunchydata.com", "reconciler kind": "BridgeCluster", "source": "kind source: /, Kind="}
2021-06-03T21:13:41.918Z	INFO	controller-runtime.manager.controller.crunchybridgeconnection	Starting EventSource	{"reconciler group": "dbaas.redhat.com", "reconciler kind": "CrunchyBridgeConnection", "source": "kind source: /, Kind="}
2021-06-03T21:13:41.918Z	INFO	controller-runtime.manager.controller.crunchybridgeinventory	Starting EventSource	{"reconciler group": "dbaas.redhat.com", "reconciler kind": "CrunchyBridgeInventory", "source": "kind source: /, Kind="}
2021-06-03T21:13:42.088Z	INFO	controller-runtime.manager.controller.bridgecluster	Starting Controller	{"reconciler group": "crunchybridge.crunchydata.com", "reconciler kind": "BridgeCluster"}
2021-06-03T21:13:42.188Z	INFO	controller-runtime.manager.controller.crunchybridgeconnection	Starting Controller	{"reconciler group": "dbaas.redhat.com", "reconciler kind": "CrunchyBridgeConnection"}
2021-06-03T21:13:42.188Z	INFO	controller-runtime.manager.controller.crunchybridgeinventory	Starting Controller	{"reconciler group": "dbaas.redhat.com", "reconciler kind": "CrunchyBridgeInventory"}
2021-06-03T21:13:42.188Z	INFO	controller-runtime.manager.controller.crunchybridgeconnection	Starting workers	{"reconciler group": "dbaas.redhat.com", "reconciler kind": "CrunchyBridgeConnection", "worker count": 1}
2021-06-03T21:13:42.188Z	INFO	controller-runtime.manager.controller.crunchybridgeinventory	Starting workers	{"reconciler group": "dbaas.redhat.com", "reconciler kind": "CrunchyBridgeInventory", "worker count": 1}
2021-06-03T21:13:42.189Z	INFO	controller-runtime.manager.controller.bridgecluster	Starting workers	{"reconciler group": "crunchybridge.crunchydata.com", "reconciler kind": "BridgeCluster", "worker count": 1}
```

### CODE REVIEW

File: apis/dbaas.redhat.com/v1alpha1/crunchybridgeinventory_types.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []CrunchyBridgeInventory `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&CrunchyBridgeInventory{}, &CrunchyBridgeInventoryList{})
+}
+
+type DatabaseProvider struct {
```
Comment (bmozaffa) [inline apis/dbaas.redhat.com/v1alpha1/crunchybridgeinventory_types.go:52]: Where/why do you need this? You should be able to use it from dbaas operator

File: config/samples/dbaas.redhat.com_v1alpha1_crunchybridgeconnection.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: dbaas.redhat.com/v1alpha1
+kind: CrunchyBridgeConnection
+metadata:
+  name: crunchybridgeconnection-sample
+spec:
+  # Add fields here
+  foo: bar
```
Comment (bmozaffa) [inline config/samples/dbaas.redhat.com_v1alpha1_crunchybridgeconnection.yaml:7]: Might be worth updating with a correct example

File: controllers/crunchybridge/bridgecluster_controller.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 Crunchy Data Solutions, Inc.
```
Comment (bmozaffa) [inline controllers/crunchybridge/bridgecluster_controller.go:2]: Accidental?

**Review by bmozaffa — CHANGES_REQUESTED** [review]
Minor comments, but looks good over all

File: apis/crunchybridge/v1/bridgecluster_types.go [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 Crunchy Data Solutions, Inc.
```
[the remark this answers is not in the document]
Comment (redhatHameed) [reply apis/crunchybridge/v1/bridgecluster_types.go]: Thanks, updated, agreed, it happened due to regenerating the controller

File: apis/dbaas.redhat.com/v1alpha1/crunchybridgeinventory_types.go (hunk 1 above)
Comment (redhatHameed) [reply apis/dbaas.redhat.com/v1alpha1/crunchybridgeinventory_types.go]: Removed. and not required here.

File: config/samples/dbaas.redhat.com_v1alpha1_crunchybridgeconnection.yaml (hunk 2 above)
Comment (redhatHameed) [reply config/samples/dbaas.redhat.com_v1alpha1_crunchybridgeconnection.yaml]: updated example as seen. 
``` apiVersion: dbaas.redhat.com/v1alpha1
kind: CrunchyBridgeConnection
metadata:
  name: crunchybridgeconnection-sample
spec:
  instanceID: hw4j2fw33zgb3mx5lpoiosj3va
  inventory:
    name: crunchybridgeinventory-sample
```

File: controllers/crunchybridge/bridgecluster_controller.go (hunk 3 above)
Comment (redhatHameed) [reply controllers/crunchybridge/bridgecluster_controller.go]: Fixed

**Review by bmozaffa — APPROVED** [review]
/lgtm

**Review by wilybrace — APPROVED** [review]

**Pull request merged**